### PR TITLE
Move versioneer and pytest configs into pyproject.toml

### PR DESCRIPTION
**Pull request by diptorupd**
We were using `setup.cfg` to define the configuration options for `versioneer` and `pytest`. Both of these can be moved into `pyproject.toml`. 

The PR moves the configurations from `setup.cfg` to `pyproject.toml` and removes `setup.cfg`.

### CODE REVIEW

**Comment by ZzEeKkAa** [comment]
`Generate coverage and push to Coveralls.io` is not triggered because of:
```yaml
name: Coverage
on:
  push:
    branches:
      - main
  pull_request:
    paths:
      - 'numba_dpex/**'
      - environment/coverage.yml
 ```
 
It simply is not triggered...

**Comment by diptorupd** [comment]
> UPD: why pytest kept in `setup.cfg` ?

I was experimenting, to see what changed to impact the coverage workflow.

File: pyproject.toml [hunk 1]
```diff
@@ -60,3 +61,24 @@ requires = [
     "versioneer-518"
 ]
 build-backend = "setuptools.build_meta"
+
+[tool.pytest.ini.options]
```
Comment (ZzEeKkAa) [inline pyproject.toml:65]:
```suggestion
[tool.pytest.ini_options]
```
https://docs.pytest.org/en/stable/reference/customize.html#pyproject-toml

File: pyproject.toml [hunk 2]
```diff
@@ -60,3 +61,24 @@ requires = [
...
+
+[tool.pytest.ini_options]
+minversion = "6.0"
+norecursedirs= [
+    ".*", "*.egg*", "build", "dist", "conda-recipe",
+]
+xfail_strict = true
+addopts = [
+    "--disable-warnings",
+    "--ignore setup.py",
```
Comment (ZzEeKkAa) [inline pyproject.toml:73]: Don't you need to make them separate strings?
```suggestion
    "--ignore", "setup.py",
```

Same for others.

Comment (ZzEeKkAa) [reply pyproject.toml:73]: or may be just for the last line, I'm not sure, but the workflow is failing

File: pyproject.toml [hunk 3]
```diff
@@ -60,3 +61,19 @@ requires = [
...
+minversion = "6.0"
+norecursedirs= [
+    ".*", "*.egg*", "build", "dist", "conda-recipe",
+]
+xfail_strict = true
+addopts = [
+    "--disable-warnings",
+    "--strict",
+    "--durations=20",
+    "-ra -vv",
```
Comment (ZzEeKkAa) [inline pyproject.toml:75]: This option set through the command line, same for disable-warnings

Comment (diptorupd) [reply pyproject.toml:75]: Removed the arguments that are specified in `pyproject.toml` from the CLI specified in the workflow when invoking `pytest`.

**Review by ZzEeKkAa — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**